src/wallet/rpcwallet.cpp

```diff
@@ -3493,33 +3521,18 @@ UniValue z_listreceivedbyaddress(const UniValue& params, bool fHelp)
             }
         },
         [&](const libzcash::SaplingPaymentAddress& addr) {
-            bool hasSpendingKey = pwalletMain->HaveSaplingSpendingKeyForAddress(addr);
-            std::set<std::pair<libzcash::SaplingPaymentAddress, uint256>> nullifierSet;
-            if (hasSpendingKey) {
-                nullifierSet = pwalletMain->GetSaplingNullifiers({addr});
-            }
-            for (const SaplingNoteEntry& entry : saplingEntries) {
-                UniValue obj(UniValue::VOBJ);
-                obj.pushKV("txid", entry.op.hash.ToString());
-                obj.pushKV("amount", ValueFromAmount(CAmount(entry.note.value())));
-                obj.pushKV("amountZat", CAmount(entry.note.value()));
-                obj.pushKV("memo", HexStr(entry.memo));
-                obj.pushKV("outindex", (int)entry.op.n);
-                obj.pushKV("confirmations", entry.confirmations);
-
-                txblock BlockData(entry.op.hash);
-                obj.pushKV("blockheight", BlockData.height);
-                obj.pushKV("blockindex", BlockData.index);
-                obj.pushKV("blocktime", BlockData.time);
-
-                if (hasSpendingKey) {
-                    obj.pushKV("change", pwalletMain->IsNoteSaplingChange(nullifierSet, entry.address, entry.op));
-                }
-                result.push_back(obj);
-            }
+            push_sapling_result(addr);
         },
         [&](const libzcash::UnifiedAddress& addr) {
-            // TODO UNIFIED
+            for (const auto& receiver : addr) {
+                std::visit(match {
+                    [&](const libzcash::SaplingPaymentAddress& addr) {
+                        push_sapling_result(addr);
+                    },
+                    [&](const auto& other) { } // TODO orchard
+
+                }, receiver);
+            }
```

Comment:
This also needs to handle transparent values received by the address, and the transparent cases should be changed to correctly report information for the transparent pool. In general, anything that works for a unified address should now also work correctly for a legacy transparent address; the objective it to no longer have users need to use two different sets of RPC methods to achieve their goals.